internal/query/access_token.go

```diff
@@ -129,26 +130,64 @@ func (q *Queries) accessTokenByOIDCSessionAndTokenID(ctx context.Context, oidcSe
 
 // checkSessionNotTerminatedAfter checks if a [session.TerminateType] event occurred after a certain time
 // and will return an error if so.
-func (q *Queries) checkSessionNotTerminatedAfter(ctx context.Context, sessionID string, creation time.Time) (err error) {
+func (q *Queries) checkSessionNotTerminatedAfter(ctx context.Context, sessionID, userID string, creation time.Time) (err error) {
 	ctx, span := tracing.NewSpan(ctx)
 	defer func() { span.EndWithError(err) }()
 
-	events, err := q.eventstore.Filter(ctx, eventstore.NewSearchQueryBuilder(eventstore.ColumnsEvent).
+	model := &sessionTerminatedModel{
+		sessionID: sessionID,
+		creation:  creation,
+		userID:    userID,
+	}
+	err = q.eventstore.FilterToQueryReducer(ctx, model)
+	if err != nil {
+		return zerrors.ThrowPermissionDenied(err, "QUERY-SJ642", "Errors.Internal")
+	}
+
+	if model.terminated {
+		return zerrors.ThrowPermissionDenied(nil, "QUERY-IJL3H", "Errors.OIDCSession.Token.Invalid")
+	}
+	return nil
+}
+
+type sessionTerminatedModel struct {
+	creation  time.Time
+	sessionID string
+	userID    string
+
+	terminated bool
+}
+
+func (s *sessionTerminatedModel) Reduce() error {
+	return nil
+}
+
+func (s *sessionTerminatedModel) AppendEvents(events ...eventstore.Event) {
+	s.terminated = len(events) > 0
+}
```

Comment:
should we append them all and then do this in the reduce?

Reply:
yes, that would be the correct way